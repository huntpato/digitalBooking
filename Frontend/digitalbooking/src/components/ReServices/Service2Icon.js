import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import styles from './Service2Icon.module.css'
import {
  faBan,
  faWifi,
  faSwimmer,
  faTv,
  faUtensils,
  faPaw,
  faCar,
  faTemperatureLow,
  faSmoking,
  faHotTub,
  faCoffee, 
  faSnowflake,
  faMugHot,
  faDumbbell,
  faWheelchair,
} from "@fortawesome/free-solid-svg-icons";

const dict = {
  wifi: faWifi,
  pool: faSwimmer,
  cabletv: faTv,
  kitchen: faUtensils,
  petsallowed: faPaw,
  parking: faCar,
  airconditioning: faTemperatureLow,
  smokingallowed: faSmoking,
  jacuzzi: faHotTub,
  breakfast: faCoffee,
  refrigerator: faSnowflake,
  microwave: faMugHot,
  gym: faDumbbell,
  accesible : faWheelchair
};

const Service2Icon = ({ icon, color }) => {
  let service;

  if (icon === undefined && !dict.hasOwnProperty(icon)) {
    service = faBan;
  } else {
    service = dict[icon.toLowerCase().replace(/ /g, "")];
  }

  return (
    <>
      <FontAwesomeIcon className= {icon} icon={service} style={color} />
    </>
  );
};

export default Service2Icon;
