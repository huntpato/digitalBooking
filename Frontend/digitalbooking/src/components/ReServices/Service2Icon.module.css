.icon{
    color: var(--color-typo);
    padding-right: 25px;
}

