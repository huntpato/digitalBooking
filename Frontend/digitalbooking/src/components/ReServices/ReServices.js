import React from "react";
import Service2Icon from "./Service2Icon";
import styles from "./ReServices.module.css";

export const ReServices = (props) => {
  const { services } = styles;
  const { locServices } = props;

  return (
    <div className={services}>
     {
      locServices.map((icon, index) =>{
        return <Service2Icon icon={icon} key={index} color= {{color: "var(--typo-color)"}}/>
      })
     }
    </div>
  );
};

ReServices.defaultProps ={

  locServices :["Wifi",
    "kitchen",
    " parking",
    "airconditioning"],
};