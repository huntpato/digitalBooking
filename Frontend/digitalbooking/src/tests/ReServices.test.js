import React from 'react';
import { shallow } from 'enzyme';
import { ReServices } from '../components/ReServices/ReServices'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';


describe('Pruebas en componente <ReServices />', ()=> {
 
    const wrapper= shallow(<ReServices services={[
        "Wifi",
        "kitchen",
        " parking",
        "airconditioning"
    ]}/>);

    test('Comprobar que <ReServices /> renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })


    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const div = wrapper.find('div');
        expect(div.hasClass('services')).toBe(true);
    })
    

})
