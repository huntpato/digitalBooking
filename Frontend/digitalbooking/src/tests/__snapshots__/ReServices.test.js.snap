// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <ReServices /> Comprobar que <ReServices /> renderiza correctamente. 1`] = `
<div
  className="services"
>
  <Service2Icon
    color={
      Object {
        "color": "var(--typo-color)",
      }
    }
    icon="Wifi"
    key="0"
  />
  <Service2Icon
    color={
      Object {
        "color": "var(--typo-color)",
      }
    }
    icon="kitchen"
    key="1"
  />
  <Service2Icon
    color={
      Object {
        "color": "var(--typo-color)",
      }
    }
    icon=" parking"
    key="2"
  />
  <Service2Icon
    color={
      Object {
        "color": "var(--typo-color)",
      }
    }
    icon="airconditioning"
    key="3"
  />
</div>
`;
